bin/getstatus.py

```diff
@@ -0,0 +1,197 @@
+#    This file is part of the HABTracker project for tracking high altitude balloons.
+#
+#    Copyright (C) 2019, Jeff Deaton (N6BA)
+#
+#    HABTracker is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    HABTracker is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with HABTracker.  If not, see <https://www.gnu.org/licenses/>.
+#
+##################################################
+
+import psutil
+import sys
+import os
+import json
+
+
+
+########################
+# getProcStatus
+# searches for the processes listed in 'query' among all procs running on the system.  It returns a list of of those found
+########################
+def getProcStatus(query: list)->list:
+
+    # where we store the processes and their status
+    procstatus = []
+
+    # Pre-populate the processes by looping through those in the 'query' list
+    for p in query:
+        procstatus.append({"process": p, "status": 0, "cmdline": None })
+
+    # Iterate over all running processes, searching for those in 'procstatus'
+    for proc in psutil.process_iter():
+
+       try:
+           # Get process detail as dictionary
+           pInfoDict = proc.as_dict(attrs=['pid', 'ppid', 'name', 'exe', 'memory_percent', 'cmdline' ])
+
+       except (psutil.NoSuchProcess, psutil.AccessDenied):
+           # we just skip over those processes that we're not allowed to see or if they no longer exist.
+           pass
+
+       else:
+           # for everyother process, we check if it's one of those that we're looking for
+           # loop over each process in procstatus checking if the current OS process is a match.  When a match is found we break out of the loop.
+           for p in procstatus:
+               if pInfoDict["name"] and pInfoDict["cmdline"]:
+                   if p["process"] in pInfoDict["name"].lower() or p["process"] in pInfoDict["cmdline"]:
+                       p["cmdline"] = pInfoDict["cmdline"]
+                       p["status"] = 1  
+                       break
+    return procstatus
+
+
+
+########################
+# isProcRunning
+# loop through the list of processes to get the status (is is running or not) of the given process name
+########################
+def isProcRunning(procs: list, name: str)->bool:
+
+    # loop through each process, looking for the process so we can determine if it's running or not
+    for p in procs:
+
+        # we want to single out if the process is running
+        if p["process"] == name:
+            return True if p["status"] > 0 else False
```

Comment:
Do you have a reason to prefer this form over the more terse 'return p["status"] > o'?

Same effect obviously so you might just consider one more readable than the other.  Just asking.

Reply:
Good catch!!  

Updated with 6c6baf2fc46eb503d6ecd6c106b8a5f65cb1aef5.